File: php/natural.php
<?php
/**
 * This script demonstrates the use of various loop
 * control structures to compute the sum of natural 
 * numbers.
 * 
 */

if($argc != 2) {
  print "ERROR: expecting an integer command line argument\n";
  exit(1);
} else if(!is_numeric($argv[1])) {
  print "ERROR: command line argument should be an integer\n";
  exit(1);
}

$n = intval($argv[1]);

$oneToTen = array(
  1 => "one",
  2 => "two",
  3 => "three",
  4 => "four",
  5 => "five",
  6 => "six",
  7 => "seven",
  8 => "eight",
  9 => "nine",
  10 => "ten"
);

//TODO: write a for-loop to add up 1 .. n and display the answer

//TODO: write a while-loop to add up 1 .. n and display the answer

//TODO: write a foreach loop to iterate over the $oneToTen array and
//      print a message as specified in the writeup

?>

File: src/main/java/unl/soc/Natural.java
package unl.soc;

/**
 * This program demonstrates the use of various loop control structures to
 * compute the sum of natural numbers.
 * 
 * @author cbourke, Lisong
 *
 */
public class Natural {

	/**
	 * Computes the sum of natural numbers 1..n using a while loop.
	 * 
	 * @return
	 */
	public static int sumWithWhile(int n) {
		// TODO: write a while-loop to compute the sum of 1..n
		return 0;
	}

	/**
	 * Computes the sum of natural numbers 1..n using a for loop.
	 * 
	 * @return
	 */
	public static int sumWithFor(int n) {
		// TODO: write a for-loop to compute the sum of 1..n
		return 0;
	}

	public static void main(String args[]) {
		if (args.length != 1) {
			System.err.println("ERROR: expecting a single integer argument");
			System.exit(1);
		}

		Integer n = null;
		try {
			n = Integer.parseInt(args[0]);
		} catch (NumberFormatException nfe) {
			System.err.println("ERROR: expecting a single integer argument");
			System.exit(1);
		}

		System.out.println("The sum of natural numbers from 1 thru "+n+" is "+sumWithWhile(n)+" using sumWithWhile()");
		
		System.out.println("The sum of natural numbers from 1 thru "+n+" is "+sumWithFor(n)+" using sumWithFor()");
		
	}
}
